tests/integration/test_transformers.py

```diff
@@ -0,0 +1,188 @@
+    if data_type is None or data_type not in generators:
+        return []
+
+    return generators[data_type]
+
+
+def _validate_input_type(input_type):
+    """Check that the transformer input type is not null."""
+    assert input_type is not None
+
+
+def _validate_transformed_data(transformer, transformed_data):
+    """Check that the transformed data is the expected dtype."""
+    expected_data_types = transformer.get_output_types()
+    transformed_dtypes = transformed_data.dtypes
+
+    for column, expected_data_type in expected_data_types.items():
+        assert column in transformed_data
+        assert transformed_dtypes[column].kind in DATA_TYPE_TO_DTYPES[expected_data_type]
+
+
+def _validate_reverse_transformed_data(transformer, reversed_data, input_dtype):
+    """Check that the reverse transformed data is the expected dtype.
+
+    Expect that the dtype is equal to the dtype of the input data.
+    """
+    expected_data_type = transformer.get_input_type()
+    assert reversed_data.dtypes[TEST_COL].kind in DATA_TYPE_TO_DTYPES[expected_data_type]
+
+
+def _validate_composition(transformer, reversed_data, input_data):
+    """Check that the reverse transformed data is equal to the input.
+
+    This is only applicable if the transformer has the composition
+    identity property.
+    """
+    if transformer.is_composition_identity():
+        pd.testing.assert_series_equal(
+            reversed_data,
+            input_data[TEST_COL],
+            check_dtype=False,
+            check_exact=False,
+            rtol=1e-03,
+        )
+
+
+def _test_transformer_with_dataset(transformer_class, input_data):
+    """Test the given transformer with the given input data.
+
+    This method verifies the transformed data's dtype, the reverse
+    transformed data (if `is_composition_identity`) and the dtype.
+
+    Args:
+        transformer_class (rdt.transformers.BaseTransformer):
+            The transformer class to test.
+        input_data (pandas.Series):
+            The data to test on.
+    """
+    transformer = transformer_class()
+    # Fit
+    transformer.fit(input_data, [TEST_COL])
+
+    # Transform
+    transformed = transformer.transform(input_data)
+    _validate_transformed_data(transformer, transformed)
+
+    # Reverse transform
+    out = transformer.reverse_transform(transformed)
+    _validate_reverse_transformed_data(transformer, out, input_data.dtypes[TEST_COL])
+
+    reversed_data = out
+    if isinstance(out, pd.DataFrame):
+        reversed_data = out[TEST_COL]
+    if isinstance(out, pd.DatetimeIndex):
+        reversed_data = out.to_series(index=pd.RangeIndex(start=0, stop=DATA_SIZE, step=1))
+    if isinstance(reversed_data, np.ndarray):
+        out = pd.Series(reversed_data)
+    if pd.api.types.is_datetime64_any_dtype(reversed_data):
+        reversed_data = reversed_data.round('us')
+
+    _validate_composition(transformer, reversed_data, input_data)
```

Comment:
If I'm reading this correctly, all these checks only matter if `transformer.is_composition_identity() == True`. Should we move this stuff to after that check inside of `_validate_composition`? Or will it be used for other checks later?

Reply:
Yes we can move it into `_validate_composition` 👍